pilot/cmd/pilot-agent/main.go

```diff
@@ -305,6 +295,39 @@ var (
 	}
 )
 
+func getPilotSAN(domain string, ns string) []string {
+	var pilotSAN []string
+	if controlPlaneAuthPolicy == meshconfig.AuthenticationPolicy_MUTUAL_TLS.String() {
+		pilotTrustDomain := role.TrustDomain
+		if len(pilotTrustDomain) == 0 {
+			if registry == serviceregistry.KubernetesRegistry &&
+				(domain == os.Getenv("POD_NAMESPACE")+".svc.cluster.local" || domain == "") {
+				pilotTrustDomain = "cluster.local"
+			} else if registry == serviceregistry.ConsulRegistry &&
+				(domain == "service.consul" || domain == "") {
+				pilotTrustDomain = ""
+			} else {
+				pilotTrustDomain = domain
+			}
+		}
+		pilotSAN = envoy.GetPilotSAN(pilotTrustDomain, ns)
+	}
+	return pilotSAN
+}
+
+func getDNSDomain(domain string) string {
+	if len(domain) == 0 {
+		if registry == serviceregistry.KubernetesRegistry {
+			domain = os.Getenv("POD_NAMESPACE") + ".svc.cluster.local"
+		} else if registry == serviceregistry.ConsulRegistry {
+			domain = "service.consul"
+		} else {
+			domain = ""
+		}
+	}
+	return domain
+}
```

Comment:
I see some redundancy in the code making the logic seemingly complex. For the above functions, can we simplify them as the following? (just an idea)
```
...
role.DNSDomain = getDNSDomain(role.DNSDomain)
trustDomain := getTrustDomain(role.DNSDomain)
pilotSAN = proxy.getPilotSAN(trustDomain, ns)
...

func getTrustDomain(dnsDomain string) []string {
	if len(role.TrustDomain) > 0 {
		return role.TrustDomain
	}
	if registry == serviceregistry.KubernetesRegistry {
		return "cluster.local"
	}
	if registry == serviceregistry.ConsulRegistry {
		return  ""
	}
	return dnsDomain
}

func getDNSDomain(domain string) string {
	if len(domain) > 0 {
		return domain
	}
	if registry == serviceregistry.KubernetesRegistry {
		return os.Getenv("POD_NAMESPACE") + ".svc.cluster.local"
	}
	if registry == serviceregistry.ConsulRegistry {
		return "service.consul"
	}
	return ""
}
```